src/charm.py

```diff
@@ -338,12 +390,46 @@ def _configure_haproxy_route(self, _: HAServiceReadyEvent) -> None:
         data = self.haproxy_route_provider.get_data(self.haproxy_route_provider.relations)
         # This is temporary as the logic to generate the haproxy config will be added later.
         logger.debug("Aggregated requirer data: %s", data)
+        self._reconcile()
 
     @validate_config_and_tls(defer=True)
     def _ensure_tls(self, _: ops.EventBase) -> None:
         """Ensure that the charm is ready to handle TLS-related events."""
         TLSInformation.validate(self)
 
+    def _get_unit_address(self) -> typing.Optional[str]:
+        """Get the current unit's address.
+
+        Returns:
+            Optional[str]: The unit's address from juju-info binding,
+                or None if the address cannot be fetched
+        """
+        network_binding = self.model.get_binding("juju-info")
+        if (
+            network_binding is not None
+            and (bind_address := network_binding.network.bind_address) is not None
+        ):
+            return str(bind_address)
+        return None
+
+    def _get_peer_units_address(self) -> list[str]:
+        """Get address of peer units.
+
+        Returns:
+            liststr]: The list of peer units address.
+        """
+        peer_units_address: list[str] = []
+        if haproxy_peer_integration := self.model.get_relation(HAPROXY_PEER_INTEGRATION):
+            for unit in haproxy_peer_integration.units:
+                if unit != self.unit:
+                    if peer_unit_address := haproxy_peer_integration.data[unit].get(
+                        "private-address"
+                    ):
+                        peer_units_address.append(peer_unit_address)
+                    else:
+                        logger.error("Cannot get address for peer unit: %s. Skipping", unit)
```

Comment:
Shouldn't this be a debug level log entry?

Reply:
updated this to warning!